pkg/util/url_test.go

```diff
@@ -4,69 +4,43 @@ import (
 	"net/url"
 	"testing"
 
-	. "github.com/smartystreets/goconvey/convey"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 func TestUrl(t *testing.T) {
-	Convey("When joining two urls where right hand side is empty", t, func() {
-		result := JoinURLFragments("http://localhost:8080", "")
-
-		So(result, ShouldEqual, "http://localhost:8080")
-	})
-
-	Convey("When joining two urls where right hand side is empty and lefthand side has a trailing slash", t, func() {
-		result := JoinURLFragments("http://localhost:8080/", "")
-
-		So(result, ShouldEqual, "http://localhost:8080/")
-	})
-
-	Convey("When joining two urls where neither has a trailing slash", t, func() {
-		result := JoinURLFragments("http://localhost:8080", "api")
-
-		So(result, ShouldEqual, "http://localhost:8080/api")
-	})
-
-	Convey("When joining two urls where lefthand side has a trailing slash", t, func() {
-		result := JoinURLFragments("http://localhost:8080/", "api")
-
-		So(result, ShouldEqual, "http://localhost:8080/api")
-	})
-
-	Convey("When joining two urls where righthand side has preceding slash", t, func() {
-		result := JoinURLFragments("http://localhost:8080", "/api")
-
-		So(result, ShouldEqual, "http://localhost:8080/api")
-	})
-
-	Convey("When joining two urls where righthand side has trailing slash", t, func() {
-		result := JoinURLFragments("http://localhost:8080", "api/")
-
-		So(result, ShouldEqual, "http://localhost:8080/api/")
-	})
-
-	Convey("When joining two urls where lefthand side has a trailing slash and righthand side has preceding slash", t, func() {
-		result := JoinURLFragments("http://localhost:8080/", "/api/")
-
-		So(result, ShouldEqual, "http://localhost:8080/api/")
-	})
+	tests := []struct {
+		base     string
```

Comment:
I would suggest adding a description field and wrapping each assertion [below](https://github.com/grafana/grafana/blob/6760a43ff9ffedea95bd39bd8ce696c8df49a4b4/pkg/util/url_test.go#L25) in a separate `t.Run` block.

Reply:
Agree with @papagian to use `t.Run`.

Reply:
✅ The description are not verbatim exactly as they were (see https://github.com/grafana/grafana/pull/26724#discussion_r469344449 ) but they should say the same thing. I also changed the name of the test. The output ends up being

```
    --- PASS: TestJoinURLFragments/where_RHS_is_empty (0.00s)
    --- PASS: TestJoinURLFragments/where_RHS_is_empty_and_LHS_has_trailing_slash (0.00s)
    --- PASS: TestJoinURLFragments/where_neither_has_trailing_slash (0.00s)
    --- PASS: TestJoinURLFragments/where_LHS_has_trailing_slash (0.00s)
    --- PASS: TestJoinURLFragments/where_LHS_and_RHS_has_trailing_slash (0.00s)
    --- PASS: TestJoinURLFragments/where_LHS_has_trailing_slash_and_RHS_has_preceding_slash (0.00s)
```

which combines brevity and readability imo